### ISSUE TITLE
Automatically apply the build-scan plugin when `--scan` is provided on the command-line

### ISSUE THREAD

**Issue by bamboo** [gradle/kotlin-dsl#532]
By taking advantage of `AutoAppliedPluginHandler` introduced in this [PR](https://github.com/gradle/gradle/pull/2832).

**Comment by gildor** [comment]
I've tried to remove build-scan from my project with `gradle-kotlin-dsl-4.3-20171010191714` but it doesn't work:
```
> ./gradlew tasks --scan

FAILURE: Build failed with an exception.

* Where:
 line: 0

* What went wrong:
Plugin [id: 'com.gradle.build-scan', version: '1.9.1', artifact: 'com.gradle:build-scan-plugin:1.9.1'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Gradle Central Plugin Repository (explicit artifact coordinates are not supported by this source)

**Comment by eskatos** [comment]
@gildor the changes mentioned in this issue are for the `0.12.1` milestone so they are not available il the version you use, see https://github.com/gradle/kotlin-dsl/releases/tag/v0.12.1

**Comment by gildor** [comment]
@eskatos I see version `gradle-kotlin-dsl-4.3-20171010191714` now on v0.12.1 release page and I copied it from this page.

**Comment by eskatos** [comment]
Ah right, sorry. The error you get is due to an issue in Gradle proper, fixed after our `v0.12.1` release by https://github.com/gradle/gradle/pull/3131. The fix will be included in Gradle 4.3-RC1 that is about to be published.
